src/internal/packager/helm/repo.go

```diff
@@ -159,8 +167,14 @@ func (h *Helm) DownloadPublishedChart(cosignKeyPath string) error {
 			chartName = h.chart.RepoName
 		}
 
-		// Perform simple chart download
-		chartURL, err = repo.FindChartInRepoURL(h.chart.URL, chartName, h.chart.Version, pull.CertFile, pull.KeyFile, pull.CaFile, getter.All(pull.Settings))
+		for _, repo := range repoFile.Repositories {
+			if repo.URL == h.chart.URL {
+				username = repo.Username
+				password = repo.Password
+			}
+		}
```

Comment:
can `repositories.yaml` contain multiple entries w/ the same URL?

Reply:
Yes 
![image](https://github.com/defenseunicorns/zarf/assets/3977569/dfa36d0a-2c71-424d-ab0e-e7acedc2cb22)

Reply:
Great point, and as far as I can tell you can have different levels of access depending on the helm repo. So someone may have access to version 6.5.0 in repo podinfo but not have access to that version with 6.5.1. Will look into changing that

Reply:
After discussions with @Racer159 we decided to move forward with this PR for now as it is a pretty unlikely edge case. We added a TODO to the code and plan on handling this case later.